### ISSUE TITLE
Review numpy API coverage

### ISSUE THREAD

**Issue by wshanks** [lmfit/uncertainties#224]
Since Uncertainties has not been updated much recently, there has been time for the functions in the unumpy module to drift relative to upstream numpy. Following on from the need to look at the numpy support because of numpy 2 (#222), it would be good to review the current unumpy interface to make sure the current functions still match numpy's versions (no change in function signature) and no functions added to numpy that a user would expect are missing. Besides that there are a couple known action items:

* numpy 2 introduced aliases for inverse trigonometric functions like `acos` for `arccos`. It would be good for unumpy to include similar aliases.
* Review if `unumpy.matrix` could be updated/deprecated. The corresponding [numpy.matrix](https://numpy.org/devdocs/reference/generated/numpy.matrix.html) has been documented as not recommended and possibly to be removed for a long time now. However, the numpy developers did not take the opportunity to remove it with numpy 2.0 and likely numpy 3.0 will not happen for a long time. Still, `unumpy.matrix` could perhaps migrate to deriving from a standard array instead of `numpy.matrix`.

**Comment by jagerber48** [comment]
#262 here is the diff for `unumpy.core`. https://github.com/lmfit/uncertainties/pull/262/files#diff-e5500f0177ef7e9fab3f620bf08f3430a76a807036f28b17e574214ad79e4b4f. I need to go through and disentangle what, if any, API change I am making. I think it was easy enough to keep support for matrices, at least in some case. I would be just as happy, actually happier, to drop support for matrices, but I think it would require at least one minor versions worth of deprecation warnings.

**Comment by jagerber48** [comment]
Ok, it's coming back to me. https://github.com/lmfit/uncertainties/pull/262/files#diff-3920e6d42d819e305be2c6e3cb5318e2bb0e7c35f62e5b25e92bd41726a869f1R165

There is a comment on this line. The way the code is now does not fully support numpy matrices. It partially supports them. Getting back support at the level of `master` would require some more work.

I vote for `uncertainties` to drop support for matrices in `4.x`. I predict everyone will agree so I'm going to make a PR for it.

**Comment by jagerber48** [comment]
Yes, the code I have now for `4.0.0` maintains a function that inverts square arrays. I think the current code can take a function that accepts one array as an argument and allow it to handle arrays with uncertainty. My new code supports code with an arbitrary number of arrays as inputs and it will detect with have uncertainty and give the appropriate result. I don't know if this generalization induces a performance hit though...

**Comment by newville** [comment]
@jagerber48  (trying to keep up - not easy!)  

With #325, is this issue now reduced to "add support for aliases introduced with numpy 2"?

**Comment by jagerber48** [comment]
#325 was a merge to `main`. So I don't think that part of this issue is closed until `main` becomes our main/released branch.

But yes, otherwise the "add support for aliases" part is outstanding. I don't feel strongly about adding that support but I'm also not strongly against it. Basically I don't want to think about it yet. That can be later 4.x release. But I didn't want the issue to stay accidentally closed.
